kong/runloop/plugins_iterator.lua

```diff
@@ -45,16 +45,31 @@ local function load_plugin_configuration(ctx,
     return ngx.exit(ngx.ERROR)
   end
 
-  if plugin ~= nil and plugin.enabled then
-    local cfg = plugin.config or {}
+  if not plugin or not plugin.enabled then
+    return
+  end
 
-    cfg.api_id      = plugin.api and plugin.api.id
-    cfg.route_id    = plugin.route and plugin.route.id
-    cfg.service_id  = plugin.service and plugin.service.id
-    cfg.consumer_id = plugin.consumer and plugin.consumer.id
+  if plugin.run_on ~= "all" then
+    if ctx.is_service_mesh_request then
+      if plugin.run_on == "first" then
+        return
+      end
 
-    return cfg
+    else
+      if plugin.run_on == "second" then
+        return
+      end
```

Comment:
I think this nested if could be simplified to:

```
if (plugin.run_on == "first" and ctx.is_service_mesh_request)
or (plugin.run_on == "second" and not ctx.is_service_mesh_request)
then
  return
end
```

Reply:
It originally was like that. Then a lot of comments about spacing, then I came with current, which is easiest to understand, imo.